collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -877,6 +882,62 @@ public void run() {
         }
     }
 
+    private void scaleDownImageIfNeeded(String path) {
+        QuestionWidget questionWidget = getWidgetWaitingForBinaryData();
+        if (questionWidget != null) {
+            Integer maxPixels = getMaxPixelsForImageIfDefined(questionWidget);
+            if (maxPixels != null) {
+                scaleDownImage(path, maxPixels);
+            }
+        }
+    }
+
+    private QuestionWidget getWidgetWaitingForBinaryData() {
+        QuestionWidget questionWidget = null;
+        for (QuestionWidget qw :  ((ODKView) currentView).getWidgets()) {
+            if (((IBinaryWidget) qw).isWaitingForBinaryData()) {
+                questionWidget = qw;
+            }
+        }
+
+        return questionWidget;
+    }
+
+    private Integer getMaxPixelsForImageIfDefined(QuestionWidget questionWidget) {
+        Integer maxPixels = null;
+        for (TreeElement attrs : questionWidget.getPrompt().getBindAttributes()) {
+            if ("max-pixels".equals(attrs.getName())) {
```

Comment:
I think we should also check the namespace here. I think there's a constant that can be used in JavaRosa or it could be added in Collect in the constants file. The important thing is to check the URI, not the prefix.